### ISSUE TITLE
bug: create temporary table inside trigger，Appear error

### ISSUE THREAD

**Issue by shangyanwen** [stoneatom/stonedb#589]
### Describe the problem

```
CREATE TABLE t1 (a INT, b INT DEFAULT 150)engine=tianmu;

delimiter |
CREATE TRIGGER t1_bi BEFORE INSERT ON t1
FOR EACH ROW 
BEGIN
  CREATE TEMPORARY TABLE t2 AS SELECT NEW.a, NEW.b;
  INSERT INTO t2(a) VALUES (10);
  INSERT INTO t2 VALUES (100, 500);
  INSERT INTO t2(a) VALUES (1000);
END|
delimiter ;

INSERT INTO t1 VALUES (1, 2);   
# error 
ERROR 1787 (HY000): Statement violates GTID consistency: CREATE TEMPORARY TABLE and DROP TEMPORARY TABLE can only be executed outside transactional context.  These statements are also not allowed in a function or trigger because functions and triggers are also considered to be multi-statement transactions.
SELECT * FROM t2;

DROP TABLE t1;
DROP TEMPORARY TABLE t2;
```

### Expected behavior

```
SELECT * FROM t2;
a  b
1  2
10  150
100  500
1000  150
```

### How To Reproduce

```
CREATE TABLE t1 (a INT, b INT DEFAULT 150)engine=tianmu;

delimiter |
CREATE TRIGGER t1_bi BEFORE INSERT ON t1
FOR EACH ROW 
BEGIN
  CREATE TEMPORARY TABLE t2 AS SELECT NEW.a, NEW.b;
  INSERT INTO t2(a) VALUES (10);
  INSERT INTO t2 VALUES (100, 500);
  INSERT INTO t2(a) VALUES (1000);
END|
delimiter ;

INSERT INTO t1 VALUES (1, 2);    
SELECT * FROM t2;

DROP TABLE t1;
DROP TEMPORARY TABLE t2;
```

### Environment

./mysqld  Ver 5.7.36_v1.0.1_beta-StoneDB for Linux on x86_64 (build-)
build information as follow: 
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: 208f79d
        Last commit time: Date:   Sat Sep 17 16:19:42 2022 +0800
        Build time: Date: Sat Sep 17 17:59:04 CST 2022

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

**Comment by duanfuxiang0** [comment]
The same problem as #580,has been fixed in #647

**Comment by shangyanwen** [comment]
Regression tests passed
